packages/plugins/federation/src/index.ts

```diff
@@ -0,0 +1,64 @@
+import { Plugin } from '@envelop/types';
+import { ApolloGateway, GatewayConfig } from '@apollo/gateway';
+import { getOperationAST, print } from 'graphql';
+import { InMemoryLRUCache, KeyValueCache } from 'apollo-server-caching';
+import { GraphQLRequestMetrics, Logger } from 'apollo-server-types';
+
+export interface FederationConfig {
+  gateway?: GatewayConfig;
+  load?: Parameters<ApolloGateway['load']>[0];
+  metrics?: GraphQLRequestMetrics;
+  cache?: KeyValueCache;
+  logger?: Logger;
+}
+
+export const useFederation = (options: FederationConfig = {}): Plugin => {
+  const {
+    gateway: gatewayConfig = {},
+    load: loadConfig = {},
+    cache = new InMemoryLRUCache(),
+    logger = console,
+    metrics = {},
+  } = options;
+  const gateway = new ApolloGateway(gatewayConfig);
+  gateway.load(loadConfig);
```

Comment:
Maybe this should be part of onPluginInit?

Reply:
I tried the alpha version locally as I was quite excited about it :)

When I used it with only the `serviceList` config param it didn't work, because  `gateway.load` is async in that case, so `gateway.schema` and `gateway.executor` are not available until the promise resolved. It looks like `onPluginInit` is called before the `gateway.load` promise is resolved, so there is an error when calling `setSchema(gateway.schema)`. Is it possible to handle async init within the envelop plugins? 🤔 

It worked perfectly when I used it with `localServiceList` or `supergraphSdl`, as those make the config static, as the gateway doesn't need to fetch the schemas asynchronously.

Reference in `@apollo/gateway` code: https://github.com/apollographql/federation/blob/f83e11ebc2a406b62d93bbed0f2a7386bb06800e/gateway-js/src/index.ts#L415-L418